crates/std_detect/src/detect/cache.rs

```diff
@@ -167,16 +128,32 @@ cfg_if::cfg_if! {
                     let _ = super::Feature::from_str(v).map(|v| value.unset(v as u32));
                 }
             }
-            CACHE.initialize(value);
+            do_initialize(value);
         }
     } else {
         #[inline]
         fn initialize(value: Initializer) {
-            CACHE.initialize(value);
+            do_initialize(value);
         }
     }
 }
 
+#[cfg(target_pointer_width = "32")]
+const CACHE_BITS: u32 = 31;
+#[cfg(target_pointer_width = "64")]
+const CACHE_BITS: u32 = 63;
```

Comment:
Could this constant either be removed or defined in terms of `mem::size_of::<usize>()` instead of using `#[cfg]`?

Reply:
Actually it could!